src/tests/nurse/test_views.py

```diff
@@ -127,6 +133,59 @@ def s3_mock():
 }
 
 
+@pytest.fixture
+def prescription(user):
+    patient = Patient.objects.create(**patient_data)
+    nurse, _ = Nurse.objects.get_or_create(user=user)
+    nurse.patients.add(patient)
+    return Prescription.objects.create(patient=patient, **prescription_data)
+
+
+@pytest.mark.django_db
+class TestSendEmailToDoctorView:
+    def test_send_email_to_doctor(self, client, prescription):
+        url = reverse_lazy("send-email-to-doctor", kwargs={"pk": prescription.id})
+
+        response = client.post(url)
+        assert response.status_code == status.HTTP_200_OK
+        assert response.data == {"message": "Email sent"}
+        assert len(mail.outbox) == 1
+        assert mail.outbox[0].to == [prescription.email_doctor]
+
+    def test_send_email_to_doctor_404(self, client):
+        url = reverse_lazy("send-email-to-doctor", kwargs={"pk": 1})
+        response = client.post(url)
+        assert response.status_code == status.HTTP_404_NOT_FOUND
+
+    def test_doctor_has_no_email(self, client, prescription):
+        prescription.email_doctor = ""
+        prescription.save()
+        url = reverse_lazy("send-email-to-doctor", kwargs={"pk": prescription.id})
+        response = client.post(url)
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
+        assert response.data == {"error": "Doctor has no email"}
+
+    def test_user_without_name(self, client, prescription, user):
+        user.first_name = ""
+        user.last_name = ""
+        user.save()
+        url = reverse_lazy("send-email-to-doctor", kwargs={"pk": prescription.id})
+        response = client.post(url)
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
+        assert response.data == {
+            "error": "Please update your profile with your first and last name"
+        }
+
+    @patch("nurse.views.send_mail", side_effect=Exception("SMTP Error"))
```

Comment:
Not needed since you use the `with patch()` context manager below

Reply:
yes , I change it with context in the function